File: tests/expected/coverage.jl

function inline_pass()
    # pass
end

function inline_ellipsis()
    # ...
end

function indexing()::Int64
    sum = 0
    a::Array{Int64} = []
    for i = 0:10-1
        push!(a, i)
        sum += a[i+1]
    end
    return sum
end

function infer_bool(code::Int64)::Bool
    return code in [1, 2, 4]
end

function show()
    a1 = 10
    b1 = 15
    b2 = 15
    @assert(b1 == 15)
    @assert(b2 == 15)
    b9 = 2
    b10 = 2
    @assert(b9 == b10)
    a2::Float64 = 2.1
    println(join([a2], " "))
    for i = 0:10-1
        println(join([i], " "))
    end
    for i = 0:2:10-1
        println(join([i], " "))
    end
    a3 = -(a1)
    a4 = a3 + a1
    println(join([a4], " "))
    t1 = a1 > 5 ? (10) : (5)
    @assert(t1 == 10)
    sum1 = indexing()
    println(join([sum1], " "))
    a5 = [1, 2, 3]
    println(join([length(a5)], " "))
    a9::Array{String} = ["a", "b", "c", "d"]
    println(join([length(a9)], " "))
    a7 = Dict("a" => 1, "b" => 2)
    println(join([length(a7)], " "))
    a8 = true
    if a8
        println(join(["true"], " "))
    else

        if a4 > 0
            println(join(["never get here"], " "))
        end
    end
    if a1 == 11
        println(join(["false"], " "))
    else

        println(join(["true"], " "))
    end
    if 1 !== nothing
        println(join(["World is sane"], " "))
    end
    println(join([true ? ("True") : ("False")], " "))
    if true
        a1 += 1
    end
    @assert(a1 == 11)
    if true
        println(join(["true"], " "))
    end
    inline_pass()
    s = "1    2"
    println(join([s], " "))
    @assert(infer_bool(1))
    _escape_quotes = " foo \"bar\" baz "
    @assert(findfirst("bbc", "aaabbccc") != Nothing)
    @assert(Bool(1))
    2
    _c1, _c2 = (1, 3)
end

function main()
    show()
end

main()

File: tests/expected/coverage.go
package main

import (
	"fmt"
	"github.com/electrious/refutil"
	iter "github.com/hgfischer/go-iter"
	"strings"
)

func InlinePass() {
	// pass
}

func InlineEllipsis() {
	// ...
}

func Indexing() int {
	var sum int = 0
	var a []int = []int{}
	for _, i := range iter.NewIntSeq(iter.Start(0), iter.Stop(10)).All() {
		a = append(a, i)
		sum += a[i]
	}
	return sum
}

func InferBool(code int) bool {
	return refutil.Contains([]int{1, 2, 4}, code)
}

func Show() {
	var a1 int = 10
	var b1 int = 15
	var b2 int = 15
	if !(b1 == 15) {
		panic("assert")
	}
	if !(b2 == 15) {
		panic("assert")
	}
	var b9 int = 2
	var b10 int = 2
	if !(b9 == b10) {
		panic("assert")
	}
	var a2 float64 = 2.1
	fmt.Printf("%v\n", a2)
	for _, i := range iter.NewIntSeq(iter.Start(0), iter.Stop(10)).All() {
		fmt.Printf("%v\n", i)
	}
	for _, i := range iter.NewIntSeq(iter.Start(0), iter.Stop(10), iter.Step(2)).All() {
		fmt.Printf("%v\n", i)
	}
	var a3 int = -(a1)
	var a4 int = (a3 + a1)
	fmt.Printf("%v\n", a4)
	var t1 int
	if a1 > 5 {
		t1 = 10
	} else {
		t1 = 5
	}
	if !(t1 == 10) {
		panic("assert")
	}
	var sum1 int = Indexing()
	fmt.Printf("%v\n", sum1)
	var a5 []int = []int{1, 2, 3}
	fmt.Printf("%v\n", len(a5))
	var a9 []string = []string{"a", "b", "c", "d"}
	fmt.Printf("%v\n", len(a9))
	a7 := map[string]int{"a": 1, "b": 2}
	fmt.Printf("%v\n", len(a7))
	var a8 bool = true
	if a8 {
		fmt.Printf("%v\n", "true")
	} else {
		if a4 > 0 {
			fmt.Printf("%v\n", "never get here")
		}
	}
	if a1 == 11 {
		fmt.Printf("%v\n", "false")
	} else {
		fmt.Printf("%v\n", "true")
	}
	if 1 != 0 {
		fmt.Printf("%v\n", "World is sane")
	}
	if true {
		fmt.Printf("%v\n", "True")
	} else {
		fmt.Printf("%v\n", "False")
	}
	if true {
		a1 += 1
	}
	if !(a1 == 11) {
		panic("assert")
	}
	if true {
		fmt.Printf("%v\n", "true")
	}
	InlinePass()
	var s string = "1    2"
	fmt.Printf("%v\n", s)
	if !(InferBool(1)) {
		panic("assert")
	}
	var _escape_quotes string = " foo \"bar\" baz "
	_ = _escape_quotes
	if !(strings.Contains("aaabbccc", "bbc")) {
		panic("assert")
	}
	if !(1 != 0) {
		panic("assert")
	}
	var _c1 int = 1
	_ = _c1
	_ = 2
	var _c2 int = 3
	_ = _c2
}

func main() {
	Show()
}
